common/oauthTokenManager.go

```diff
@@ -765,6 +796,11 @@ func (credInfo *OAuthTokenInfo) GetTokenCredential() (azcore.TokenCredential, er
 	if credInfo.PSCred {
 		return credInfo.GetPSContextCredential()
 	}
+
+	if credInfo.WorkloadIdentity {
```

Comment:
Q: is there a reason we continue to use several booleans here instead of a simple enum/switch like you've changed it to in login.go?

Reply:
oh whoops this was an oversight from before I changed to the string version. I'll clean this up.